lib/hammer_cli_foreman/compute_profile.rb

```diff
@@ -0,0 +1,422 @@
+        params['id'] =  params['compute_attribute']['id']
+        params['compute_attribute']['vm_attrs'] = params['compute_attribute']['vm_attrs'].merge(vm_attr.merge(option_compute_attributes || {} ))
+
+        if option_interface
+          params['compute_attribute']['vm_attrs']['interfaces_attributes'] = interface_attr.merge({option_interface_id => option_interface})
+        end
+        if option_volume
+          unless option_volume_id
+            params['compute_attribute']['vm_attrs']['volumes_attributes'] = volume_attr.merge({option_volume_id => option_volume})
+          end
+        end
+        params
+      end
+      success_message _('Compute profile attributes updated.')
+      failure_message _('Could not update the compute profile attributes')
+
+      build_options :without => :id
+    end
+
+    class AddInterface < HammerCLIForeman::UpdateCommand
+      command_name _('add-interface')
+      resource :compute_attributes
+      desc _('Add interface for Compute Profile')
+      option '--set-values', 'SET VALUES', _('Interface parameters, should be comma separated list of values.'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      extend_help do |h|
+        h.section _('Provider Interface specific options') do |h|
+          ::HammerCLIForeman::ComputeProfile.compute_resources.each do |name, provider|
+            h.section name do |h|
+              if defined?(provider.interface_create_help)
+                provider.interface_create_help(h)
+              end
+            end
+          end
+        end
+      end
+
+      def validate_options
+        super
+        validator.any(:option_compute_profile_id,:option_compute_profile_name).required
+        validator.any(:option_compute_resource_id,:option_compute_resource_name).required
+        validator.all(:option_set_values).required
+      end
+
+      def request_params
+        params = HammerCLIForeman::ComputeProfile.get_params(options)
+        interface_attr =  params['compute_attribute']['vm_attrs']['interfaces_attributes'] || {}
+        params['id'] = params['compute_attribute']['id']
+        params['compute_attribute']['vm_attrs']['interfaces_attributes'] = interface_attr.merge({Time.now.to_i.to_s => option_set_values})
+        params
+      end
+
+      success_message _('Interface was created.')
+      failure_message _('Could not create interface')
+
+      build_options :without => :id
+    end
+
+    class UpdateInterface < HammerCLIForeman::UpdateCommand
+      command_name _('update-interface')
+      resource :compute_attributes
+
+      desc _('Update compute profile interface')
+
+      option '--set-values', 'SET VALUES', _('Interface parameters, should be comma separated list of values.'),
+             :required => true,
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface-id', 'INTERFACE_ID', _('Interface id.'),
```

Comment:
ditto "dot at the end"

Reply:
i don't really understand where to put the dot here